api-server/controllers/controllersv1/deployment.go

```diff
@@ -421,7 +426,9 @@ func (c *deploymentController) Delete(ctx *gin.Context, schema *GetDeploymentSch
 	if err != nil {
 		return nil, err
 	}
-	return transformersv1.ToDeploymentSchema(ctx, deployment)
+	deploymentSchema, err := transformersv1.ToDeploymentSchema(ctx, deployment)
+	go tracking.TrackDeploymentEvent(ctx, deploymentSchema, tracking.YataiDeploymentDelete)
```

Comment:
qq: why does this has to be a goroutine?

Reply:
didn't want to make the tracking requests blocking which is why we are spinning off a goroutine to every tracking request